sdk/core/azure-core/inc/azure/core/uuid.hpp

```diff
@@ -35,7 +36,7 @@ namespace Azure { namespace Core {
      * @brief Gets Uuid as a string.
      * @details A string is in canonical format (8-4-4-4-12 lowercase hex and dashes only).
      */
-    std::string ToString() const;
+    std::string ToString();
```

Comment:
Why not `const std::string& ToString() const;`? Part of the point of caching this is that you don't have to copy the std::string value in the `ToString()` method.

Reply:
No more caching of the string atm, so marking this as resolved.
If we change the underlying representation of Uuid to `std::string`, in the future, we can revisit this suggestion.

Reply:
IIRC the currently checked in version is marked as const, why mark it as mutable?

Reply:
Or maybe a different question: Why block calling `ToString()` on a `const Uuid`?

Reply:
The diff you are seeing in your comment is outdated. I left `ToString()` as const.